strawberryfields/tdm/tdmprogram.py

```diff
@@ -83,63 +84,136 @@ def validate_measurements(circuit, N):
     return spatial_modes
 
 
-def input_check(args, copies):
+def input_check(args):
     """Checks the input arguments have consistent dimensions.
 
     Args:
         args (Sequence[Sequence]): sequence of sequences specifying the value of the parameters
-        copies (int): number of times the circuit should be run
-
     """
     # check if all lists are of equal length
     param_lengths = [len(param) for param in args]
     if len(set(param_lengths)) != 1:
         raise ValueError("Gate-parameter lists must be of equal length.")
 
-    # check copies
-    if not isinstance(copies, int) or copies < 1:
-        raise TypeError("Number of copies must be a positive integer.")
-
 
-def _get_mode_order(num_of_values, N):
+def _get_mode_order(num_of_values, N, timebins):
     """Get the order by which the modes were measured"""
```

Comment:
(Out of scope:) I find that it would be nice to have an example here and add the arguments, return values to the docstring.

Reply:
I would say it's fine leaving this as it is, since the mode order wouldn't be something that visible to the user. It's simply a hidden helper function collecting which modes that were measured in the simulator (nothing to do with the actual modes being returned).

Reply:
Right. Although docstrings are not written entirely for users, I personally found it pretty difficult to parse this function.